packages/core/src/integrations/supabase.ts

```diff
@@ -0,0 +1,493 @@
+    if (typeof authOperation === 'function') {
+      supabaseClientInstance.auth[operation] = instrumentAuthOperation(authOperation);
+    }
+  });
+
+  AUTH_ADMIN_OPERATIONS_TO_INSTRUMENT.forEach((operation: AuthAdminOperationName) => {
+    const authAdminOperation = auth.admin[operation];
+    if (typeof authAdminOperation === 'function') {
+      supabaseClientInstance.auth.admin[operation] = instrumentAuthOperation(authAdminOperation, true);
+    }
+  });
+}
+
+function instrumentSupabaseClientConstructor(SupabaseClient: unknown): void {
+  if (instrumented.has(SupabaseClient)) {
+    return;
+  }
+
+  (SupabaseClient as unknown as SupabaseClientConstructor).prototype.from = new Proxy(
+    (SupabaseClient as unknown as SupabaseClientConstructor).prototype.from,
+    {
+      apply(target, thisArg, argumentsList) {
+        const rv = Reflect.apply(target, thisArg, argumentsList);
+        const PostgrestQueryBuilder = (rv as PostgrestQueryBuilder).constructor;
+
+        instrumentPostgrestQueryBuilder(PostgrestQueryBuilder as unknown as new () => PostgrestQueryBuilder);
+
+        return rv;
+      },
+    },
+  );
+}
+
+// This is the only "instrumented" part of the SDK. The rest of instrumentation
+// methods are only used as a mean to get to the `PostgrestFilterBuilder` constructor itself.
+function instrumentPostgrestFilterBuilder(PostgrestFilterBuilder: PostgrestFilterBuilder['constructor']): void {
+  if (instrumented.has(PostgrestFilterBuilder)) {
+    return;
+  }
+
+  instrumented.set(PostgrestFilterBuilder, {
+    then: (PostgrestFilterBuilder.prototype as unknown as PostgrestProtoThenable).then,
+  });
+
+  (PostgrestFilterBuilder.prototype as unknown as PostgrestProtoThenable).then = new Proxy(
+    (PostgrestFilterBuilder.prototype as unknown as PostgrestProtoThenable).then,
+    {
+      apply(target, thisArg, argumentsList) {
+        const operations = AVAILABLE_OPERATIONS;
+        const typedThis = thisArg as PostgrestFilterBuilder;
+        const operation = extractOperation(typedThis.method, typedThis.headers);
+
+        if (!operations.includes(operation)) {
+          return Reflect.apply(target, thisArg, argumentsList);
+        }
+
+        if (!typedThis?.url?.pathname || typeof typedThis.url.pathname !== 'string') {
+          return Reflect.apply(target, thisArg, argumentsList);
+        }
+
+        const pathParts = typedThis.url.pathname.split('/');
+        const table = pathParts.length > 0 ? pathParts[pathParts.length - 1] : '';
+        const description = `from(${table})`;
+
+        const queryItems: string[] = [];
+        for (const [key, value] of typedThis.url.searchParams.entries()) {
+          // It's possible to have multiple entries for the same key, eg. `id=eq.7&id=eq.3`,
+          // so we need to use array instead of object to collect them.
+          queryItems.push(translateFiltersIntoMethods(key, value));
+        }
+
+        const body: Record<string, unknown> = {};
```

Comment:
```suggestion
        const body: Record<string, unknown> = Object.create(null);
```

to prevent prototype pollution.